chat.py

```diff
@@ -36,6 +36,10 @@ def generate(
     assert max_seq_length > T
     buffer = max((len(tokens) for tokens in stop_tokens), default=0)
 
+    if model.device.type == "xla":
+        import torch_xla.core.xla_model as xm
+        xm.mark_step()
```

Comment:
What's the reasoning behind calling this before forward is called (before the loop)?

Reply:
Thanks for pointing this out, I don't think this is needed. I will leave the import there but will remove the mark_step().

Reply:
@gkroiz imports are cached in Python, it can be inside the loop